astropy/coordinates/distances.py

```diff
@@ -105,10 +105,13 @@ def __new__(cls, value=None, unit=None, z=None, cosmology=None,
                 raise ValueError('Should given only one of `value`, `z` '
                                  'or `distmod` in Distance constructor.')
 
+            import astropy.cosmology.units as cu
+
             if cosmology is None:
                 from astropy.cosmology import default_cosmology
                 cosmology = default_cosmology.get()
 
+            z = u.Quantity(z, cu.redshift).value
```

Comment:
Doesn't `cosmology.luminosity_distance` already do the conversion?

Reply:
It does indeed. Rectified.